nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/pom.xml

```diff
@@ -147,6 +147,12 @@
             <version>2.2.1</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.nifi</groupId>
+            <artifactId>nifi-solr-client-api</artifactId>
+            <version>1.8.0-SNAPSHOT</version>
+            <scope>compile</scope>
```

Comment:
This should be provided scope, using compile scope would mean the nifi-solr-client-api JAR would be packaged into two different NARs (the API NAR and the processors NAR) and at runtime it should only be in the API NAR. 

This relates to the other comment that the nifi-solr-nar pom needs to be updated to change its parent from standard services to the nifi-solr-client-api-nar.

I'm not in a state to try it right now, but I would suspect that running a full build of this branch you would be able to create a SolrClientService, but not assign it in one of the Solr processors.

Reply:
Was able to try it out and you are right that it is working, but currently this is only because the service impl happens to be in the same NAR as the processors. If someone implemented their own SolrClientService in another NAR then it wouldn't be able to be used in the current state.

If you just change the above dependency to provided and then change nifi-solr-nar/pom.xml where it has:
```
        <dependency>
            <groupId>org.apache.nifi</groupId>
            <artifactId>nifi-standard-services-api-nar</artifactId>
            <version>1.8.0-SNAPSHOT</version>
            <type>nar</type>
        </dependency>
```
To
```
        <dependency>
            <groupId>org.apache.nifi</groupId>
            <artifactId>nifi-solr-client-api-nar</artifactId>
            <version>1.8.0-SNAPSHOT</version>
            <type>nar</type>
        </dependency>
```
That should make it work correctly for both cases.